homeassistant/components/aurora_abb_powerone/__init__.py

```diff
@@ -42,6 +42,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         except AuroraError as error:
             if "No response after" in str(error):
                 raise ConfigEntryNotReady("No response (could be dark)") from error
+            else:
+                raise error
```

Comment:
Shouldn't we raise `ConfigEntryNotReady` here too? If there's some other connection error we want to retry, right? What different kinds of errors can we encounter?

Reply:
No response is really the only one we want to retry on.  All of the others are wrong serial port or could not open port etc.  In those cases I don't think retrying is going to help.

Reply:
How will those errors be resolved? Does the user need to do something?

Reply:
They probably need to try different settings, but since this is for yaml import, then it is likely that the that the old settings worked before.  If they dont, the user either needs to type new ones into configuration.yaml and retry the import (restart hass I suppose).  Or just delete those entries it and do it through the UI (where they will see the errors in the config flow messages).

Reply:
Were you thinking we could present the failed setup in the integrations page and allow the user to adjust the configuration using whatever was imported from configuration.yaml as the start point?

Reply:
I'm thinking maybe we should make a connection attempt in the config flow during import and abort the import if there are other errors than no response. Only on no response error will we let the entry be created anyway and reach here. Then we should minimize the risk of these errors here.

I think we should return False here for these errors and write a comment why.

Reply:
Hmm, what if the connection to the device between Home Assistant and the device fails somehow. Won't that also generate errors and in that case we do want to retry later?

Reply:
I think these could all be the same error.  It is an 'old school' RS485 interface so once you know that you are talking to a valid physical com port on the HA machine, then once you start trying to send data, the lines will toggle up and down.  Whether anything listens or replies is or is even wired at all is unknown until a response is received.  All of the following would lead to 'no response':
- Darkness
- Inverter broken
- RS485 interface broken
- Backwards wiring
- Cable disconnected, cut or shorted out.
- Baud rate/parity mismatch etc.

Not sure what else there could be.  If it was a TCP/IP link or similar you could get timeouts, routing errors, stack errors etc, but those don't exist on hardwired RS485.

Reply:
Ok, let's go with `return False` here and a comment. And would be good to do the connection check during config flow import too, and abort if there are other errors than no response. The latter can be a follow up, if it's a lot of work.

Reply:
Ok.  I have looked around the other integrations for a bit of context.  I get the impression that we aren't supposed to raise arbitrary exceptions from async_setup_entry().  So if there is an OSError or similar due to invalid com port, I should be catching all of those here.  So I'm now looking at swapping:
```
        except AuroraError as error:
            if "No response after" in str(error):
                raise ConfigEntryNotReady("No response (could be dark)") from error
            raise error
```
for 
```
        except AuroraError as error:
            if "No response after" in str(error):
                raise ConfigEntryNotReady("No response (could be dark)") from error
            raise ConfigEntryNotReady(
                "Failed to connect to inverter: %s" % error
            ) from error
        except OSError as error:
            if error.errno == 19:  # No such device.
                raise ConfigEntryNotReady(
                    "Failed to connect to inverter: no such COM port"
                ) from error
            raise error
        else:
...
```
I suppose that ConfigEntryNotReady() is an ok response.  It will retry, and maybe someone could have replugged in the COM port by that time, in which case ok to continue.

Does that sound good?

Reply:
Yes, it's ok to raise `ConfigEntryNotReady` to retry, but we should not raise other arbitrary errors. Just return False in that case.

Reply:
Done. Converted to raise ConfigEntryNotReady for both exceptions that we catch.